include/ginkgo/core/matrix/csr.hpp

```diff
@@ -1042,11 +1076,50 @@ class Csr : public EnableLinOp<Csr<ValueType, IndexType>>,
         this->make_srow();
     }
 
+    /**
+     * Creates a CSR matrix from already allocated (and initialized) row
+     * pointer, column index and value arrays.
+     *
+     * @note This is the same as the previous constructor but with a default
+     *       strategy.
+     */
+    template <typename ValuesArray, typename ColIdxsArray,
+              typename RowPtrsArray>
+    Csr(std::shared_ptr<const Executor> exec, const dim<2>& size,
+        ValuesArray&& values, ColIdxsArray&& col_idxs, RowPtrsArray&& row_ptrs)
+        : Csr{exec,
+              size,
+              std::forward<ValuesArray>(values),
+              std::forward<ColIdxsArray>(col_idxs),
+              std::forward<RowPtrsArray>(row_ptrs),
+              Csr::make_default_strategy(exec)}
+    {}
+
     void apply_impl(const LinOp* b, LinOp* x) const override;
 
     void apply_impl(const LinOp* alpha, const LinOp* b, const LinOp* beta,
                     LinOp* x) const override;
 
+    // TODO: This provides some more sane settings. Please fix this!
+    static std::shared_ptr<strategy_type> make_default_strategy(
+        std::shared_ptr<const Executor> exec)
+    {
+        auto cuda_exec = std::dynamic_pointer_cast<const CudaExecutor>(exec);
+        auto hip_exec = std::dynamic_pointer_cast<const HipExecutor>(exec);
+        auto dpcpp_exec = std::dynamic_pointer_cast<const DpcppExecutor>(exec);
+        std::shared_ptr<strategy_type> new_strategy;
+        if (cuda_exec) {
+            new_strategy = std::make_shared<automatical>(cuda_exec);
+        } else if (hip_exec) {
+            new_strategy = std::make_shared<automatical>(hip_exec);
+        } else if (dpcpp_exec) {
+            new_strategy = std::make_shared<classical>();
```

Comment:
should it be automatical or classical?

Reply:
I think it depends on your experience. If I remember, you said classical was almost always faster? Or are there some cases where load_balance was useful as well?

Reply:
load_balance has pretty limited benefit on intel DG1.
automatical does the same thing, which almost always chooses classical from the condition.
when it is on device executor, we use automatical by default. (dpcpp tend to choose classical in automatical strategy)

Reply:
So you want the decision to be left to automatical. I will use it then. As a side comment, I think it's fine to change the default strategies now that we have this, if we think it provides better performance then there shouldn't be a problem (e.g. switching to sparse lib or another).